### PR TITLE
cam6_3_124: Fix memory issue, default PE layout for FMTHIST

### LINKED ISSUE #812: Increase PE layout for MT runs
`###` Issue Type

Infrastructure Update

### Issue Description

MT runs out-of-the-box error out while writing restart files.

@tilmes reported that the following PE layout worked for her and preliminary testing indicates it does solve the problem:

```
</entry>
    <entry id="NTASKS">
      <type>integer</type>
      <values>
        <value compclass="ATM">-50</value>
        <value compclass="CPL">-50</value>
        <value compclass="OCN">-50</value>
        <value compclass="WAV">1</value>
        <value compclass="GLC">1</value>
        <value compclass="ICE">-50</value>
        <value compclass="ROF">-50</value>
        <value compclass="LND">-50</value>
        <value compclass="ESP">1</value>
      </values>
      <desc>number of tasks for each component</desc>
    </entry>
    <entry id="NTASKS_PER_INST">
      <type>integer</type>
      <values>
        <value compclass="ATM">1800</value>
        <value compclass="OCN">1800</value>
        <value compclass="WAV">1</value>
        <value compclass="GLC">1</value>
        <value compclass="ICE">1800</value>
        <value compclass="ROF">1800</value>
        <value compclass="LND">1800</value>
        <value compclass="ESP">1</value>
      </values>
      <desc>Number of tasks per instance for each component. DO NOT EDIT: Set automatically by case.setup based on NTASKS, NINST and MULTI_DRIVER</desc>
    </entry>
    <entry id="NTHRDS">
      <type>integer</type>
      <values>
        <value compclass="ATM">1</value>
        <value compclass="CPL">1</value>
        <value compclass="OCN">1</value>
        <value compclass="WAV">1</value>
        <value compclass="GLC">1</value>
        <value compclass="ICE">1</value>
        <value compclass="ROF">1</value>
        <value compclass="LND">1</value>
        <value compclass="ESP">1</value>
      </values>
      <desc>number of threads for each task in each component</desc>
    </entry>


```
### Will this change answers?

No

### Will you be implementing this yourself?

Any CAM SE can do this

### LINKED ISSUE #841: Introduce regression tests for FLTHIST and FMTHIST
### What is the feature/what would you like to discuss?

Need to track down bug described in #812 with FMTHIST configuration

Add regression tests for FLTHIST and FMTHIST

### Is there anyone in particular you want to be part of this conversation?

@jtruesdal 

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Any CAM SE can do this

### PR DESCRIPTION
**Pull request by fvitt**
closes #812 
closes #841

### CODE REVIEW

**Comment by cacraigucar** [comment]
ERP test gave a floating point exception in wetdep.  My test case is at:
glade/scratch/cacraig/ERP_D_Ln9_Vnuopc.ne30pg3_ne30pg3_mg17.FMTHIST.cheyenne_gnu.cam-outfrq9s.20230810_165939_t5efa4

1:Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
1:
1:Backtrace for this error:
109:#0  0x2b376d377bff in ???
109:#1  0x11bfc2b in __wetdep_MOD_wetdepa_v2
109:    at /glade/scratch/cacraig/cam_unstr_emis_fix/src/chemistry/aerosol/wetdep.F90:522
109:#2  0xc74c03 in __aero_model_MOD_aero_model_wetdep
109:    at /glade/scratch/cacraig/cam_unstr_emis_fix/src/chemistry/modal_aero/aero_model.F90:1543

**Review by cacraigucar — CHANGES_REQUESTED** [review]
Thank

**Comment by fvitt** [comment]
> ERP test gave a floating point exception in wetdep. My test case is at: glade/scratch/cacraig/ERP_D_Ln9_Vnuopc.ne30pg3_ne30pg3_mg17.FMTHIST.cheyenne_gnu.cam-outfrq9s.20230810_165939_t5efa4
> 
> 1:Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation. 1: 1:Backtrace for this error: 109:#0 0x2b376d377bff in ??? 109:#1 0x11bfc2b in __wetdep_MOD_wetdepa_v2 109: at /glade/scratch/cacraig/cam_unstr_emis_fix/src/chemistry/aerosol/wetdep.F90:522 109:#2 0xc74c03 in __aero_model_MOD_aero_model_wetdep 109: at /glade/scratch/cacraig/cam_unstr_emis_fix/src/chemistry/modal_aero/aero_model.F90:1543

The ERP test should pass with my latest changes

File: cime_config/config_pes.xml [hunk 1]
```diff
@@ -333,6 +333,33 @@
   </grid>
   <grid name="a%ne30" >
     <mach name="cheyenne">
+      <pes pesize="any" compset="_CAM%DEV%MT%GHGMAM4_">
```
[the remark this answers is not in the document]
Comment (fvitt) [reply cime_config/config_pes.xml]: The default PE layouts for derecho seem to work okay

File: cime_config/testdefs/testlist_cam.xml [hunk 2]
```diff
@@ -1589,6 +1589,24 @@
...
+  <test compset="FLTHIST" grid="ne30pg3_ne30pg3_mg17" name="SMS_D_Ln9" testmods="cam/outfrq9s">
+    <machines>
+      <machine name="cheyenne" compiler="intel" category="aux_cam"/>
+    </machines>
+    <options>
+      <option name="wallclock">00:20:00</option>
+      <option name="comment">"CAM7 dev low top ~40 km"</option>
+    </options>
+  </test>
+  <test compset="FMTHIST" grid="ne30pg3_ne30pg3_mg17" name="SMS_Ln9" testmods="cam/outfrq9s">
```
[the remark this answers is not in the document]
Comment (fvitt) [reply cime_config/testdefs/testlist_cam.xml]: changed as suggested

File: cime_config/testdefs/testlist_cam.xml [hunk 3]
```diff
@@ -1589,6 +1589,24 @@
       <option name="wallclock">00:20:00</option>
     </options>
   </test>
+  <test compset="FLTHIST" grid="ne30pg3_ne30pg3_mg17" name="SMS_D_Ln9" testmods="cam/outfrq9s">
+    <machines>
+      <machine name="cheyenne" compiler="intel" category="aux_cam"/>
```
[the remark this answers is not in the document]
Comment (fvitt) [reply cime_config/testdefs/testlist_cam.xml]: changed as suggested

**Review by cacraigucar — APPROVED** [review]

**Comment by cacraigucar** [comment]
> > ERP test gave a floating point exception in wetdep. My test case is at: glade/scratch/cacraig/ERP_D_Ln9_Vnuopc.ne30pg3_ne30pg3_mg17.FMTHIST.cheyenne_gnu.cam-outfrq9s.20230810_165939_t5efa4
> > 1:Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation. 1: 1:Backtrace for this error: 109:#0 0x2b376d377bff in ??? 109:#1 0x11bfc2b in __wetdep_MOD_wetdepa_v2 109: at /glade/scratch/cacraig/cam_unstr_emis_fix/src/chemistry/aerosol/wetdep.F90:522 109:#2 0xc74c03 in __aero_model_MOD_aero_model_wetdep 109: at /glade/scratch/cacraig/cam_unstr_emis_fix/src/chemistry/modal_aero/aero_model.F90:1543
> 
> The ERP test should pass with my latest changes

I should have pointed out that I was able to run the ERP test which previously failed, and it does indeed run properly.  Thanks @fvitt!!

File: cime_config/testdefs/testlist_cam.xml [hunk 4]
```diff
@@ -1589,6 +1589,42 @@
       <option name="wallclock">00:20:00</option>
     </options>
   </test>
+  <test compset="FLTHIST" grid="ne30pg3_ne30pg3_mg17" name="SMS_D_Ln9" testmods="cam/outfrq9s">
+    <machines>
+      <machine name="cheyenne" compiler="intel" category="prealpha"/>
+    </machines>
+    <options>
+      <option name="wallclock">00:20:00</option>
+      <option name="comment">"CAM7 dev low top ~40 km"</option>
```
Comment (nusbaume) [inline cime_config/testdefs/testlist_cam.xml:1598]: I don't think we need the quotations inside the comment text itself, e.g.:

```suggestion
      <option name="comment">CAM7 dev low top ~40 km</option>
```

The same holds for the other test description comments below as well.

File: src/dynamics/se/dyn_comp.F90 [hunk 5]
```diff
@@ -763,8 +763,10 @@ subroutine dyn_init(dyn_in, dyn_out)
           (hvcoord%hyam(:)+hvcoord%hybm(:))*hvcoord%ps0,km_sponge_factor,&
           kmvis_ref,kmcnd_ref,rho_ref)
 
-     write(iulog,*) "Molecular viscoity and thermal conductivity reference profile"
-     write(iulog,*) "k, p, z, km_sponge_factor, kmvis_ref/rho_ref, kmcnd_ref/(cp*rho_ref):"
+     if (masterproc) then
+        write(iulog,*) "Molecular viscoity and thermal conductivity reference profile"
```
Comment (nusbaume) [inline src/dynamics/se/dyn_comp.F90:767]: Typo here (`viscoity` -> `viscosity`).

**Review by nusbaume — APPROVED** [review]
Looks good!  I found a couple small things but nothing that would require a re-review.

**Pull request merged**